### PR TITLE
refactor: [TECH-605][TECH-655] Checkbox and Switch migration

### PR DESCRIPTION
**Pull request by simonadomnisoru**
Migrate Material-UI checkbox to DHIS2/UI checkbox 

There are 4 cases where `Checkbox` is used. For the 4th case, the filter with one option, the `Checkbox` was migrated to `RadioBox` button and the "Remove filter" and enable/disable CTA buttons were added. 

1) In the modal to select which columns to disply in the events table. 
Screenshot 2021-08-05 at 11 08 43

2) In the filters where you can select multiple options
Screenshot 2021-08-05 at 11 09 06

3) In true only fitlers
Screenshot 2021-08-05 at 13 50 02

4) In the filters where you can only select one option 
Screenshot 2021-08-05 at 11 09 16

### CODE REVIEW

File: src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js [hunk 1]
```diff
@@ -89,17 +91,19 @@ export const withButtons = () => (InnerComponent: React.ComponentType<any>) =>
                                 muiButtonRef={this.setUpdateButtonInstance}
                                 primary
                                 onClick={this.handleUpdateClick}
+                                disabled={disabledUpdate}
                             >
-                                {i18n.t('Apply')}
+                                {i18n.t('Update')}
```
Comment (huocha) [inline src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js:96]: Can you explain why this text label changes from `Apply` to `Update`?

File: src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js [hunk 2]
```diff
@@ -89,17 +91,19 @@ export const withButtons = () => (InnerComponent: React.ComponentType<any>) =>
...
                         </div>
                         <Button
                             dataTest="list-view-filter-cancel-button"
                             muiButtonRef={this.setCloseButtonInstance}
                             secondary
                             onClick={onClose}
+                            disabled={disabledReset}
                         >
-                            {i18n.t('Cancel')}
+                            {i18n.t('Reset filter')}
```
Comment (huocha) [inline src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js:106]: Also why this changes from `Cancel` to `Reset filter`?

**Review by huocha — COMMENTED** [review]
hey @simonadomnisoru, the code LGTM! can you attach the screenshot of previous state and current state of these checkbox to understand better

File: src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js (hunk 1 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js:96]: The text change is part of the new filters [redesign and functionality](https://docs.google.com/document/d/1BGxFZ6G64KcAS95Cmh0-ab5BzPbBGN5_AIh-2V5cVJA/edit#heading=h.pzj6mk7lxht4) . Thank you!

File: src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js (hunk 2 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/ListView/Filters/Contents/withButtons.js:106]: The text change is part of the new filters [redesign and functionality](https://docs.google.com/document/d/1BGxFZ6G64KcAS95Cmh0-ab5BzPbBGN5_AIh-2V5cVJA/edit#heading=h.pzj6mk7lxht4) . Thank you!

**Comment by huocha** [comment]
thanks @simonadomnisoru LGTM!

**Review by huocha — APPROVED** [review]

**Review by JoakimSM — APPROVED** [review]
LGTM
